hazelcast/src/main/java/com/hazelcast/spi/CanCancelOperations.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright (c) 2008-2016, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.spi;
+
+import com.hazelcast.nio.Address;
+
+/**
+ * Implemented by a service that can cancel its operations.
+ */
+public interface CanCancelOperations {
```

Comment:
Actually the current state is broken with respect to `Future` contract: 
> After this method returns, subsequent calls to `isDone()` will always return `true`.
The new state is correct because `Future` is not job-control API; by `cancel`ing it you simply waive your interest in the eventual return value. Taking no action to percipitate task completion is by the book.